pkg/ingester/errors.go

```diff
@@ -50,15 +57,18 @@ func newErrorWithHTTPStatus(err error, code int) errorWithStatus {
 }
 
 func (e errorWithStatus) Error() string {
-	return e.status.String()
+	return e.status.Message()
 }
 
 func (e errorWithStatus) Unwrap() error {
 	return e.err
 }
 
-func (e errorWithStatus) GRPCStatus() *status.Status {
-	return e.status
+func (e errorWithStatus) GRPCStatus() *grpcstatus.Status {
```

Comment:
Note for reviewer: this is needed to ensure that gRPC engine will be able to recognizer `errorWithStatus` by calling `grpc.status.FromError()`.